bindings/c/src/common.rs

```diff
@@ -851,23 +851,49 @@ impl node_builder {
     }
 }
 
-#[repr(C)]
 pub struct tree {
-    pub root: node_id,
+    _private: [u8; 0],
+}
+
+impl CastPtr for tree {
+    type RustType = Tree;
 }
 
+impl BoxCastPtr for tree {}
+
 impl tree {
     #[no_mangle]
-    pub extern "C" fn accesskit_tree_new(root: node_id) -> tree {
-        tree { root }
+    pub extern "C" fn accesskit_tree_new(root: node_id) -> *mut tree {
+        let t = Tree::new(root.into());
+        BoxCastPtr::to_mut_ptr(t)
     }
-}
 
-impl From<tree> for Tree {
-    fn from(tree: tree) -> Self {
-        Self {
-            root: tree.root.into(),
-        }
+    #[no_mangle]
+    pub extern "C" fn accesskit_tree_free(t: *mut tree) {
+        drop(box_from_ptr(t));
+    }
+
+    #[no_mangle]
+    pub extern "C" fn accesskit_tree_set_app_name(t: *mut tree, app_name: *const c_char) {
```

Comment:
For consistency with other types, having get and clear functions for each property would be nice. I'm also not a big fan of one-letter variables, especially arguments.

Reply:
I would just go with `tree`.

Reply:
A, I thought having a struct and variable with the same name would fail, but turns out it doesn't.

I'm not very up to speed with macros yet. DO you suggest using macros or should I just create the getters and clearers manually?

Reply:
The macros we currently have can't be reused, and it would not be worth to define new ones just for a struct. Hand-writting these functions seem the best option to me.